graphql_compiler/schema_transformation/rename_query.py

```diff
@@ -131,18 +169,56 @@ def leave_selection_set(
     def enter_field(
         self, node: FieldNode, key: Any, parent: Any, path: List[Any], ancestors: List[Any]
     ) -> Union[FieldNode, VisitorAction]:
-        """Rename root vertex fields."""
+        """Rename fields."""
         # For a Field to be a root vertex field, it needs to be the first level of
         # selections (fields in more nested selections are normal fields that should not be
         # modified)
         # As FragmentDefinition is not allowed, the parent of the selection must be a query
         # As a query may not start with an inline fragment, all first level selections are
         # fields
         if self.selection_set_level == 1:
-            renamed_node = self._rename_name(node)
-            if renamed_node is node:  # Name unchanged, continue traversal
-                return None
-            else:  # Name changed, return new node, `visit` will make shallow copies along path
-                return renamed_node
+            self.current_type_name.append(node.name.value)
+        else:
+            # Entered a regular field and we want to find its type
+            current_type_name = self.current_type_name[-1]
+            current_type = self.schema.get_type(current_type_name)
+            if not current_type:
+                raise AssertionError(
+                    f"Current type is {current_type_name} which doesn't exist in the schema. This "
+                    f"is a bug."
+                )
+            if not current_type.ast_node:
+                raise AssertionError(
+                    f"Current type {current_type_name} should have non-null field ast_node, which "
+                    f"contains information such as the current type's fields. However, ast_node "
+                    f"was None. This is a bug."
+                )
+            if not isinstance(
+                current_type.ast_node, (ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode)
+            ):
+                raise AssertionError(
+                    f"Current type {current_type_name}'s ast_node field should be an "
+                    f"ObjectTypeDefinitionNode. However, the actual type was "
+                    f"{type(current_type.ast_node)}. This is a bug."
+                )
+            current_type_fields = current_type.ast_node.fields
+            for field_node in current_type_fields:
+                # Unfortunately, fields is a list instead of some other datastructure so we actually
+                # have to loop through them all.
+                if field_node.name.value == node.name.value:
+                    field_type_name = get_ast_with_non_null_and_list_stripped(
```

Comment:
Linting fails here because this function is untyped. I tried adding type hints so that it takes a `TypeNode` argument (based on how this function is used) and returns a `Node`, but it'd involve some more issues where the code assumes the `Node` objects have a `name` field so I wanted to hold off on this in case there was a cleaner way to do this.

Reply:
Yeah, it's not great and it's a tricky thing to fix. In the interest of speed, let's allow (only) this file to call untyped code by modifying the mypy.ini. Let me know if you'd like a hand with that.

Reply:
This ended up being a bigger change than expected but that's just because `typing_copilot` merged a lot of rules into one now that we're allowing untyped calls again